src/services/db.js

```diff
@@ -0,0 +1,27 @@
+import mongoose from 'mongoose';
+import AppError from '../errors/AppError.js';
+import { getMongoUri } from '../config/env.js';
+
+export async function connectDB() {
+  try {
+    await mongoose.connect(getMongoUri());
+
+    mongoose.connection.on('error', (err) => {
+      throw new AppError('Database connection error', 500, true);
+    });
+
+    mongoose.connection.on('disconnected', () => {
+      throw new AppError('Database disconnected', 500, true);
```

Comment:
try not to hardcode the status codes and use the the constants from the errorManagment object